### PR TITLE
mat: make EigenSym satisfy Matrix

### PR DESCRIPTION
**Pull request by vladimir-ch**
Please take a look.

### CODE REVIEW

File: mat/eigen_test.go [hunk 1]
```diff
@@ -153,25 +154,26 @@ func TestSymEigen(t *testing.T) {
 		var es EigenSym
 		ok := es.Factorize(test.mat, true)
 		if !ok {
-			t.Errorf("bad factorization")
+			t.Errorf("bad test")
```
Comment (kortschak) [inline mat/eigen_test.go:157]: These might want some identifying mark to explain which case was bad — pointless here with only one case, but below might be good. Not crucial.

**Review by kortschak — DISMISSED** [review]

File: mat/eigen_test.go (hunk 1 above)
Comment (vladimir-ch) [reply mat/eigen_test.go]: Done.

**Comment by vladimir-ch** [comment]
@kortschak Can you please approve again? I pushed changes by mistake.

**Review by kortschak — APPROVED** [review]
GitHub's PR review UX is horrible.

**Pull request merged**